src/fdosecrets/widgets/AccessControlDialog.cpp

```diff
@@ -0,0 +1,227 @@
+/*
+ *  Copyright (C) 2013 Francois Ferrand
+ *  Copyright (C) 2017 KeePassXC Team <team@keepassxc.org>
+ *  Copyright (C) 2020 Aetf <aetf@unlimitedcodeworks.xyz>
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "AccessControlDialog.h"
+#include "ui_AccessControlDialog.h"
+
+#include "fdosecrets/widgets/RowButtonHelper.h"
+
+#include "core/Entry.h"
+
+#include <QWindow>
+
+#include <utility>
+
+AccessControlDialog::AccessControlDialog(QWindow* parent, const QList<Entry*>& items, const QString& name, uint pid)
+    : m_ui(new Ui::AccessControlDialog())
+    , m_model(new EntryModel(items))
+{
+    if (parent) {
+        // Force the creation of the QWindow, without this windowHandle() will return nullptr
+        winId();
+        auto window = windowHandle();
+        Q_ASSERT(window);
+        window->setTransientParent(parent);
+    }
+    setWindowFlags(windowFlags() | Qt::WindowStaysOnTopHint);
+
+    m_ui->setupUi(this);
+
+    connect(m_ui->cancelButton, &QPushButton::clicked, this, [this]() { done(DenyAll); });
+    connect(m_ui->allowButton, &QPushButton::clicked, this, [this]() { done(AllowSelected); });
+    connect(m_ui->itemsTable, &QTableView::clicked, m_model.data(), &EntryModel::toggleCheckState);
+    connect(m_ui->rememberCheck, &QCheckBox::clicked, this, &AccessControlDialog::rememberChecked);
+    connect(this, &QDialog::finished, this, &AccessControlDialog::dialogFinished);
+
+    m_ui->rememberMsg->setCloseButtonVisible(false);
+    m_ui->rememberMsg->setMessageType(MessageWidget::Information);
+
+    m_ui->appLabel->setText(m_ui->appLabel->text().arg(name).arg(pid));
+
+    m_ui->itemsTable->setModel(m_model.data());
+    installWidgetItemDelegate<DenyButton>(m_ui->itemsTable, 2, [this](QWidget* p, const QModelIndex& idx) {
+        auto btn = new DenyButton(p, idx);
+        connect(btn, &DenyButton::clicked, this, &AccessControlDialog::denyEntryClicked);
+        return btn;
+    });
```

Comment:
I personally don't see a purpose for the "Deny for this program" button in this context. We are not saving the decision permanently like the browser dialog does and it replicates the same behavior that the checkbox provides. For simplicity, we should just use the checkboxes for this dialog.

Reply:
Makes sense. I'll disable it behind a flag, but not removing it, given we're going to unify/reuse the dialog for browser.